### ISSUE TITLE
Triggers app reload

### ISSUE THREAD

**Issue by entorb** [NathanChen198/streamlit-cookies-controller#10]
After quite some headache, I found `streamlit_cookies_controller` to trigger an unexpected app reload (and breaking my msal Azure login).

Running this minimal app
```python
import streamlit as st
from streamlit.logger import get_logger
logger = get_logger("Cookie Test 1")
logger.info("Start")
s = "Value"
st.write(s)
logger.info("End")
```
I see 2 expected entries in the log:
```
2024-12-18 11:15:13.437 INFO    Cookie Test 1   Start
2024-12-18 11:15:13.774 INFO    Cookie Test 1   End
```

Introducing streamlit_cookies_controller
```python
import streamlit as st
from streamlit.logger import get_logger
from streamlit_cookies_controller import CookieController
logger = get_logger("Cookie Test 2")
logger.info("Start")
cc = CookieController()
s = "Value"
# cc.set("myCookie", s)
# s = cc.get("myCookie")
st.write(s)
logger.info("End")

```
the app is running twice:
```
2024-12-18 11:16:24.988 INFO    Cookie Test 2   Start
2024-12-18 11:16:28.110 INFO    Cookie Test 2   End
2024-12-18 11:16:28.473 INFO    Cookie Test 2   Start
2024-12-18 11:16:28.475 INFO    Cookie Test 2   End
```
Interestingly, this is not the case when pressing "R" in the app to reload the page.

This problem breaks my msal login via `initiate_auth_code_flow` / `acquire_token_by_auth_code_flow`, please fix.

Browsers used: Edge 131.0.2903.99 and Firefox 133.0.3

**Comment by gihad** [comment]
Same here, did you find a solution or alternative?

**Comment by klimma1508** [comment]
Hi,

I ran into the same issue and, based on my research, it appears to be an inherent limitation of how Streamlit handles client‑side state—though I’m not 100% sure, so please correct me if I’m off. When you “set” a cookie from Python, Streamlit injects a `<script>` tag that writes to document.cookie in the browser context. Because there’s no built‑in mechanism to push that change back into the running app, the only way to surface the new cookie value on the Python side is by forcing a full page reload.

In short, the extra reload isn’t a bug in your code—it seems to be just how Streamlit bridges between Python and the browser’s cookie store.
